SiteOps/ApproveRules.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python
+
+import sys
+import os
+import getopt
+import pprint
+from rucio.client.client import Client
+from rucio.common.exception import (RuleNotFound, AccessDenied)
+
+# default mode
+auto = False
+silent = False
+verbose = False
+
+def usage():
+    me = os.path.basename(sys.argv[0])
+    print "usage:"
+    print
+    print "%s [-f <file>] [-h] [-s] [-a] <id> ..."%(me)
+    print
+    print "    <file>: rules, one in each row, beginning with rule id as the first token"
+    print "      <id>: rule id"
+    print
+    print "    -h this message"
+    print "    -s silent mode, no output; without -a, it still prompt for action"
+    print "    -a auto mode, automatically approve the ones that can be approved"
+    print "       by default, without '-a', %s asks for confirmation"%(me)
+    print "    -v verbose mode, overridden by -s"
+    print
+    print "    ** without arguments, %s takes input from stdin"%(me)
+    print "       and gives warning if '-a' is not set"
+
+opt, args = getopt.getopt(sys.argv[1:], "f:hasv")
```

Comment:
Please use argparse. getopt is super old

https://docs.python.org/3.8/library/argparse.html#module-argparse

Reply:
Yes. getopt is old and I am from old school. Replaced getopt with argparse. :-)